tools/controller.js

```diff
@@ -98,13 +98,22 @@ Controller.prototype.onToggleLineBreakpoint = function (line, clickedSpan) {
   this.view.updateLineBreakpoint(breakpoint);
 };
 
-Controller.prototype.onToggleMessageSendBreakpoint = function (e) {
-  dbgLog("onToggleMessageSendBreakpoint");
+Controller.prototype.messageBreakpointReceiver = function (sectionId) {
+  dbgLog("--messageBreakpointReceiver");
+  
+  var sourceSection = this.dbg.getSection(sectionId),
+  breakpoint    = this.toggleBreakpoint(sectionId,
+      function (source) { return new SendBreakpoint(source, sourceSection, "receiver"); });
 
-  var sectionId   = e.currentTarget.id,
-    sourceSection = this.dbg.getSection(sectionId),
-    breakpoint    = this.toggleBreakpoint(sectionId,
-      function (source) { return new SendBreakpoint(source, sourceSection); });
+  this.view.updateSendBreakpoint(breakpoint);
+};
+
+Controller.prototype.messageBreakpointSender = function (sectionId) {
+  dbgLog("--messageBreakpointSender");
+
+  var sourceSection = this.dbg.getSection(sectionId),
+  breakpoint    = this.toggleBreakpoint(sectionId,
+      function (source) { return new SendBreakpoint(source, sourceSection, "sender"); });
 
```

Comment:
Hmm I don't like this code.
This is code duplication in those two methods, no?
The only difference is "receiver" vs "sender", right?

Would be good to have a method that takes that as a parameter.

Also, the method you replaced was name "onToggle", why is this naming not preserved for these methods?
This indicates that these methods are UI toggling event handlers.